websockets/echo/src/main.rs

```diff
@@ -4,18 +4,30 @@
 
 use actix_files::NamedFile;
 use actix_web::{middleware, web, App, Error, HttpRequest, HttpResponse, HttpServer, Responder};
-use actix_web_actors::ws;
+use ractor::Actor;
 
 mod server;
-use self::server::MyWebSocket;
+use self::server::{AsMessage, MyWebSocket};
 
 async fn index() -> impl Responder {
     NamedFile::open_async("./static/index.html").await.unwrap()
 }
 
 /// WebSocket handshake and start `MyWebSocket` actor.
 async fn echo_ws(req: HttpRequest, stream: web::Payload) -> Result<HttpResponse, Error> {
-    ws::start(MyWebSocket::new(), &req, stream)
+    let (res, session, stream) = actix_ws::handle(&req, stream)?;
+
+    let (actor, _handle) = Actor::spawn(None, MyWebSocket, session).await.unwrap();
+
+    actix_web::rt::spawn(async move {
```

Comment:
we also have utilities in a companion crate which adds the actor protection on top of stream handling, so you can benefit from supervision and cleanly handling streams. https://docs.rs/ractor_actors/latest/ractor_actors/streams/pump/fn.spawn_stream_pump.html

but that's just an aside.